### PR TITLE
Replace guava and commons-lang with JDK internals

### PR DESCRIPTION
**Pull request by lorban**

### CODE REVIEW

File: src/main/java/com/bpodgursky/jbool_expressions/ExprUtil.java [hunk 1]
```diff
@@ -49,9 +49,9 @@
 
   public static <K> Expression<K> stripNegation(Or<K> or, And<K> internalAnd, Expression<K> omitFromAnd){
     Expression<K>[] childrenNew = ExprUtil.allExceptMatch(or.expressions, internalAnd);
-    List<Expression<K>> newChildren = Lists.newArrayList(childrenNew);
+    List<Expression<K>> newChildren = new ArrayList<>(Arrays.asList(childrenNew));
```
Comment (bpodgursky) [inline src/main/java/com/bpodgursky/jbool_expressions/ExprUtil.java]: Couldn't this just be 

````
List<Expression<K>> newChildren = Arrays.asList(childrenNew);
````
and avoid a copy?

File: src/main/java/com/bpodgursky/jbool_expressions/And.java [hunk 2]
```diff
@@ -29,7 +32,7 @@ private And(Expression<K>[] children, Comparator<Expression> comparator) {
 
   public String toString() {
     if (!cachedStringRepresentation.isPresent()) {
-      cachedStringRepresentation = Optional.of("(" + StringUtils.join(expressions, " & ") + ")");
+      cachedStringRepresentation = Optional.of("(" + String.join(" & ", Arrays.stream(expressions).map(Object::toString).collect(Collectors.toList())) + ")");
```
Comment (bpodgursky) [inline src/main/java/com/bpodgursky/jbool_expressions/And.java]: just being fiddly as get familiar with some of the java 8 streaming syntax, but 
````
cachedStringRepresentation = Optional.of("(" +Arrays.stream(expressions).map(Object::toString).collect(Collectors.joining("&"))+")");
````
is apparently also available

File: src/main/java/com/bpodgursky/jbool_expressions/ExprUtil.java (hunk 1 above)
Comment (bpodgursky) [reply src/main/java/com/bpodgursky/jbool_expressions/ExprUtil.java]: Oh weird, it's returning a different fixed size ArrayList.  That is annoying

Comment (bpodgursky) [reply src/main/java/com/bpodgursky/jbool_expressions/ExprUtil.java]: nvm, ok, I get it.  it's just a view over the array.  so it's only one copy.  ignore this whole chain.

File: src/main/java/com/bpodgursky/jbool_expressions/And.java (hunk 2 above)
Comment (AlbertGevorgyan) [reply src/main/java/com/bpodgursky/jbool_expressions/And.java]: Yes, that is correct. You can also include the prefix and suffix in Collectors.joining call.

**Pull request merged**